src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -474,30 +476,33 @@ private File generateTargetFile(String originalFilename, String targetFolder, Bo
      * @param keepParentFolder    if true also creates the resources with the folder they are currently in (e.g. current/parent/* -> new/parent/*)
      * @throws IOException if the copying operation fails.
      */
-    public void copyResources(Resource[] resources, String prefix, String targetDirectoryPath, Boolean keepParentFolder) throws IOException {
+
+    public void copyResources(Resource[] resources, String prefix, Path targetDirectoryPath, Boolean keepParentFolder) throws IOException {
+
         for (Resource resource : resources) {
             // Replace windows separator with "/"
             String fileUrl = java.net.URLDecoder.decode(resource.getURL().toString(), StandardCharsets.UTF_8).replaceAll("\\\\", "/");
             // cut the prefix (e.g. 'exercise', 'solution', 'test') from the actual path
             int index = fileUrl.indexOf(prefix);
 
-            String targetFilePath = keepParentFolder ? fileUrl.substring(index + prefix.length()) : "/" + resource.getFilename();
+            // We use prefix.length() + 1 to also cut the "/" after the prefix
+            String targetFilePath = keepParentFolder ? fileUrl.substring(index + prefix.length() + 1) : resource.getFilename();
```

Comment:
Can the `prefix` be empty and the `fileUrl` start without `/`?

Reply:
It is currently not forbidden. This could lead to cutting the first letter from a path.